### PR TITLE
Add missing APIs for analytics collector

### PR DESCRIPTION
**Pull request by rolandkakonyi**
This PR adds missing API for the analytics integration:
- `CustomData.experimentName` property
- `Player.analyticsCollector.addSourceMetadata` API and related `SourceMetadata` type

### CODE REVIEW

**Comment by matamegger** [comment]
@rolandkakonyi with the recently introduced Player/Analytics bundling I guess we will change the implementation of some of the introduced functions in a next step. right?

**Comment by rolandkakonyi** [comment]
> @rolandkakonyi with the recently introduced Player/Analytics bundling I guess we will change the implementation of some of the introduced functions in a next step. right?

@matamegger yes, that is the plan, but that will require bigger changes around the analytics integration in my opinion.
This is to provide the missed APIs to help the resolution of #176.

File: android/src/main/java/com/bitmovin/player/reactnative/AnalyticsModule.kt [hunk 1]
```diff
@@ -126,6 +126,17 @@ class AnalyticsModule(private val context: ReactApplicationContext) : ReactConte
...
     }
 
+    @ReactMethod
+    fun addSourceMetadata(nativeId: NativeId, playerId: NativeId?, json: ReadableMap?) {
+        uiManager()?.addUIBlock { _ ->
+            playerModule()?.getPlayer(playerId)?.source?.let { playerSource ->
+                JsonConverter.toAnalyticsSourceMetadata(json)?.let { sourceMetadata ->
+                    collectors[nativeId]?.addSourceMetadata(playerSource, sourceMetadata)
+                }
+            }
```
Comment (matamegger) [inline android/src/main/java/com/bitmovin/player/reactnative/AnalyticsModule.kt:136]: Let's make this a bit easier to digest by removing one layer please.

```suggestion
            val playerSource = playerModule()?.getPlayer(playerId)?.source ?: return
            JsonConverter.toAnalyticsSourceMetadata(json)?.let { sourceMetadata ->
                collectors[nativeId]?.addSourceMetadata(playerSource, sourceMetadata)
            }
```

**Review by matamegger — APPROVED** [review]
good addition. Seems to be inline with the existing code.

Please consider my suggestion.

File: android/src/main/java/com/bitmovin/player/reactnative/AnalyticsModule.kt (hunk 1 above)
Comment (rolandkakonyi) [reply android/src/main/java/com/bitmovin/player/reactnative/AnalyticsModule.kt:136]: Thanks for the suggestion!
Taken!

**Pull request merged**